controllers/provider-azure/pkg/controller/controlplane/valuesprovider.go

```diff
@@ -289,3 +289,10 @@ func getInfraNames(infraStatus *apisazure.InfrastructureStatus) (string, string,
 
 	return nodesSubnet.Name, nodesRouteTable.Name, nodesSecurityGroup.Name, nil
 }
+
+// GetStorageClassesChartValues returns the values for the storage classes chart applied by this actuator.
+func (vp *valuesProvider) GetStorageClassesChartValues(ctx context.Context, cp *extensionsv1alpha1.ControlPlane, cluster *extensionscontroller.Cluster) (map[string]interface{}, error) {
```

Comment:
As discussed and raised by @vpnachev  you probably can remove this code and check in the chart for `.Capabilities.Kube.Version` instead of `.Values.kubernetesVersion`.

Reply:
Yes, it is already used in the very same chart https://github.com/gardener/gardener-extensions/blob/e76686203ce1dd0b1fd8b4ef92a33ebe8dae5914/controllers/provider-azure/charts/internal/shoot-storageclasses/templates/_helpers.tpl#L2